doc/specs/#1564 - Settings UI/design.md

```diff
@@ -0,0 +1,112 @@
+
+![Keyboard page modal](./keyboard-modal.png)
+
+If the command they select has additional arguments/actions, the modal will dynamically size as arguments/actions are added.
+
+![Keyboard page modal add new arguments](./keyboard-modal-add.png)
+
+![Keyboard page modal arguments](./keyboard-modal-args.png)
+
+## Settings layout
+
+Below is the list of all settings on their respective pages in the settings UI. The title row aligns with the navigation view on the left of the UI. Bolded headers in those columns align with top nav on the page.
+
+| General - Startup | General - Interaction | General - Rendering | Appearance - Global | Appearance - Color Schemes | Profiles - Global | Profiles - Enumerate profiles | Profiles - Add new |
+| ---------------- | --------------------- | ------------------- | ------------------- | -------------------------- | ----------------- | ----------------------------- | ------------------ |
+| Default profile (dropdown) | Copy after selection is made (checkbox) | Software rendering (checkbox) | Theme (radio) | Name (text box) | **General** | **General** | **General** | **General** |
+| Launch on startup (checkbox) | Copy formatting (checkbox) | Screen redrawing (checkbox) | Show/Hide the title bar (checkbox) | Cursor color (color picker) | Command line (text box) | Scrollbar visibility (radio) | Scrollbar visibility (radio) |
+| Launch size (radio) | Word delimiters (text box) | | Show terminal title in title bar (checkbox) | Selection background (color picker) | Starting directory (browse button) | Command line (browse button) | Command line (browse button) |
+| Launch position (text box) | Window resize behavior (checkbox) | | Always show tabs (checkbox) | Background (color picker) | Icon (browse button) | Starting directory (browse button) | Starting directory (browse button) |
+| Columns on first launch (number picker) | | | Tab width mode (radio) | Foreground (color picker) | Tab title (text box) | Name (text box) | Name (text box) |
+| Rows on first launch (number picker) | | | Hide close all tabs popup (checkbox) | Black (color picker) | Scrollbar visibility (radio) | Icon (browse button) | Icon (browse button) |
+| Automatically create new profiles when new shells are installed (checkbox) | | | | Blue (color picker) | **Appearance** | Tab title (text box) | Tab title (text box) |
+| | | | | Cyan (color picker) | Font face (text box) | **Appearance** | **Appearance** |
+| | | | | Green (color picker) | Font size (number picker) | Retro terminal effects (checkbox) | Retro terminal effects (checkbox) |
+| | | | | Purple (color picker) | Font weight (text box) | Font face (text box) | Font face (text box) |
+| | | | | Red (color picker) | Padding (text box) | Font size (number picker) | Font size (number picker) |
+| | | | | White (color picker) | Cursor shape (radio) | Font weight (text box) | Font weight (text box) |

```

Comment:
Proposal: Font Weight should be a dropdown, and a numeric value can _only_ be input via the JSON.

Reply:
Oh oops I think I removed that from the other file and forgot to update this one.